### ISSUE TITLE
Fix Timezone error on MetricsJsonSerializer

### ISSUE THREAD

**Issue by viniciusccarvalho** [spring-cloud/spring-cloud-stream#927]
The Json serializer is set to write dates using GMT as base, but the DateFormatter is not set for that, causing wrong values to be written as string representations.

**Comment by olegz** [comment]
Hmm. . . Actually based on the preliminary testing it writes date using local time zone which could obviously have implication if such timezone is not conveyed  properly to a subscriber in the different timezone.
There is also seem to be a bug in the actual format String
`"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"`
This simply prints 'Z' at the end
```
{"name":"Hello","value":123.0,"timestamp":"2017-04-24T10:28:55.902Z"}
```
I believe that it should be `"yyyy-MM-dd'T'HH:mm:ss.SSS Z"` to effectively convey the time zone of the date
```
{"name":"Hello","value":123.0,"timestamp":"2017-04-24T10:30:03.574 -0400"}
```

Back to the original issue. . . another approach would be to *always* print date as GMT. But I do not think it's the right idea and could lead to many mistakes due to misinterpretation.

**Comment by viniciusccarvalho** [comment]
Adding a timezone to the formatter fix the issue, I've tested it already locally.  The formatters can read the timezone with a `Z` at the end as well as the time difference to GMT. What needs to be done is adding a timezone to the formatter that will be able to write according to GMT - The intention is to _always_ write in GMT.

**Comment by olegz** [comment]
Well, then we have two issues
1. The 'Z' in the current format is meaningless since it's quoted, so the quotes needs to be removed to get any meaning out of it.
2. If we add GMT timezone to the `DateFormat`, then what is the purpose of the 'Z' at the end? It will always be `0000`. Perhaps you meant `z` (lowercase). That with combination of `df.setTimeZone(TimeZone.getTimeZone("GMT"));` will accomplish what I believe you are looking for `{"name":"Hello","value":123.0,"timestamp":"2017-04-24T15:15:59.261 GMT"}`

**Comment by viniciusccarvalho** [comment]
Unit tests suggest the opposite. The Z needs to be quoted to be parsed correctly. And there's several examples on stackoverflow of usage of 'Z' quoted. 
I have a unit tests to try it with 'Z' and then setting and reading from GMT. It's not meaningless because it can hint other parsers (outside of java land) that this time is actually at zulu time.

**Comment by mbogoevici** [comment]
@olegz 
The `Z` in an ISO 8601 encoded date  (which is intended to be used hear) means `UTC` - see https://en.wikipedia.org/wiki/ISO_8601#Time_zone_designators , so the current approach of hardcoding the letter would be correct if the date that would be written would actually represent the UTC time. We can do it in many other ways but I think that the original intent of always sending stuff in UTC and marking it as such is correct. Simply adding setting the correct timezone on the formatter should fix the issue.

**Comment by olegz** [comment]
Ok, done a bit more reading, so. . .
I think what's you need is the 'X' option which directly corresponds to "ISO 8601 time zone". 
```
static DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS X");
```
In this case IF DateFormat is not preset with the timezone (`df.setTimeZone(TimeZone.getTimeZone("GMT"));) the output will look as such:
```
{"name":"Hello","value":123.0,"timestamp":"2017-04-24T12:36:11.367 -04"}
```
and if DateFormat is set with the timezone then you'll see the Z at the end
```
{"name":"Hello","value":123.0,"timestamp":"2017-04-24T12:36:11.367 Z"}
```

**Comment by viniciusccarvalho** [comment]
merged via 8fc0cf1a0c1b8f8edcf542734048e59e42c74383
